chore(android): Update projects to use Java 11 (#8543)
keymanapp/keyman: docs/build/windows.md

* Ant
* Gradle
* Maven

> * OpenJDK 11 (for 17.0)
> * OpenJDK 8 (for stable-16.0)

Reviewer: ```suggestion
* OpenJDK 11
```

We don't need to maintain docs for older versions in the alpha branch -- you'd look at the `stable-16.0` branch to read docs for that version.

```ps1
# Elevated PowerShell
choco install android-sdk android-studio ant gradle maven
# optionally install sdk images
sdkmanager "system-images;android-33;google_apis;armeabi-v7a"
sdkmanager --update
sdkmanager --licenses
```

* Run Android Studio once after installation to install additional components
  such as emulator images and SDK updates.

**Required environment variables**:
* [`JAVA_HOME`](#java_home)

> 

Reviewer: ```suggestion

**Optional environment variables**:
* [`JAVA_HOME_11`](#java_home)

```

**Requirements**:

> * emscripten 2.0.23+

Reviewer: ```suggestion
* emscripten 3.1.32
```

We should make this change for all platforms

> * node.js 18+

Reviewer: ```suggestion
* node.js 18
```

`-DUSERDEFINES`. You can also set an environment variable `USERDEFINES=1` to get
the same result.

> ### JAVA_HOME

Reviewer: I wonder if this whole section should be in a non-platform-specific doc?

This environment variable tells Gradle what version of Java to use for building Keyman for Android.

> OpenJDK 8 is used to build stable-16.0 while OpenJDK 11 is used for master.
> It's recommended to install both and set the environment variables to:
> ```bat
> SET JAVA_HOME_8="path to OpenJDK 8"
> SET JAVA_HOME_11="path to OpenJDK 11"
> SET JAVA_HOME="$JAVA_HOME_8"
> ```

Reviewer: ````suggestion

**Multiple versions of Java:** If you need to build Keyman for Android 16.0 or older versions, you can set `JAVA_HOME_11` to the OpenJDK 11 path and `JAVA_HOME` to the OpenJDK 8 path. This will build both versions correctly from command line. But note that you do need to update your `JAVA_HOME` env var to the associated version before opening Android Studio and loading any Android projects. `JAVA_HOME_11` is mostly used by CI.

```bat
SET JAVA_HOME="path to OpenJDK 8"
SET JAVA_HOME_11="path to OpenJDK 11"
```
````
